src/renderer/components/data-settings/data-settings.js

```diff
@@ -735,6 +735,95 @@ export default Vue.extend({
       showToast(this.$t('Settings.Data Settings.Subscriptions have been successfully exported'))
     },
 
+    importYouTubeHistory: async function() {
+      const options = {
+        properties: ['openFile'],
+        filters: [
+          {
+            name: this.$t('Settings.Data Settings.History File'),
+            extensions: ['json']
+          }
+        ]
+      }
+
+      const response = await showOpenDialog(options)
+      if (response.canceled || response.filePaths?.length === 0) {
+        return
+      }
+      let textDecode
+      try {
+        textDecode = await readFileFromDialog(response)
+      } catch (err) {
+        const message = this.$t('Settings.Data Settings.Unable to read file')
+        showToast(`${message}: ${err}`)
+        return
+      }
+
+      const historyData = JSON.parse(textDecode)
+      const filterPredicate = item => item.products.includes('YouTube') &&
+                                      item.activityControls.includes('YouTube watch history') &&
```

Comment:
This would only work if the account doing the takeout is an english account.

Reply:
I see. On my case, I wanted to get rid of the history of watching Google Ads. How bad do you think would it be to have them also on the history? Google Ads reports look like this to me:

```json
{
  "header": "YouTube",
  "title": "Watched Umweltschutz. Anschauen.",
  "titleUrl": "https://www.youtube.com/watch?v\u003dJXzjCTvb30Q",
  "description": "Watched at 7:03pm",
  "time": "2022-12-06T18:03:39.962Z",
  "products": ["YouTube"],
  "details": [{
    "name": "From Google Ads"
  }],
  "activityControls": ["Web \u0026 App Activity", "YouTube watch history", "YouTube search history"]
}
```

Maybe we can use this details array to filter out. How does the Google Ads appear in French?

Reply:
I'm having trouble recreating the export (it seems to only be exporting the .html now even though I'm changing it to json) but it should be structured the same just different text so filtering on the details array would work

Reply:
You can also check if the titleUrl exists (not sure if it's present in the search history) or if the titleUrl is a youtube url

Reply:
FreeTube doesn't have any search history at the moment, so it wouldn't make sense to import it, as you would need to design the entire functionality around it first.

Reply:
The bigger issue with probing YouTube or Invidious for each video is that you are very likely to get ratelimited making that volume of requests (ratelimits take a while to disappear sometimes multiple days), as I presume people will have a large history lists.

Reply:
Yes, I thought that also. That's why I didn't even try and have set default values, in order for the UI to look a little better. Completely agree with @absidue.

What about the idea of using the localization infrastructure to have translatable strings for stripping out that suffix from the title?

Reply:
> FreeTube doesn't have any search history at the moment, so it wouldn't make sense to import it, as you would need to design the entire functionality around it first.

What is this, then? I thought that we had some kind of history, thus the option to clear it.

<img width="220" alt="image" src="https://user-images.githubusercontent.com/326831/207652620-7a0507a7-a920-4782-8572-4dd5b9834954.png">